app/src/main/kotlin/com/wire/android/services/OngoingCallService.kt

```diff
@@ -131,17 +133,42 @@ class OngoingCallService : Service() {
         scope.cancel()
     }
 
-    private fun generateForegroundNotification(callName: String, conversationId: String, userId: UserId) {
+    private fun generateForegroundNotification(
+        callName: String,
+        conversationId: String,
+        userId: UserId
+    ) {
         appLogger.i("$TAG: generating foregroundNotification...")
-        val notification: Notification = callNotificationManager.builder.getOngoingCallNotification(callName, conversationId, userId)
-        startForeground(CALL_ONGOING_NOTIFICATION_ID, notification)
+        val notification: Notification = callNotificationManager.builder.getOngoingCallNotification(
+            callName,
+            conversationId,
+            userId
+        )
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            startForeground(
+                CALL_ONGOING_NOTIFICATION_ID,
+                notification,
+                ServiceInfo.FOREGROUND_SERVICE_TYPE_PHONE_CALL or ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE
```

Comment:
can this be replaces with `        ServiceCompat.startForeground(this, CALL_ONGOING_NOTIFICATION_ID, notification, (ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE or ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE))` ? if so then there will be no need  to check the sdk version